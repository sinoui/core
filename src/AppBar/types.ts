export interface AppBarStatusTypes {
  /**
   * 突出模式
   */
  prominent?: boolean;
  /**
   * 密集模式
   */
  dense?: boolean;
  /**
   * 固定模式
   */
  fixed?: boolean;
  /**
   * 可收缩
   */
  short?: boolean;
  /**
   * 一直处于收缩的状态的应用栏
   */
  shortCollapsed?: boolean;
}

export interface AppBarProps extends AppBarStatusTypes {
  /**
   * 应用栏左侧图标
   */
  navigationIcon?: React.ReactNode;
  /**
   * 应用栏标题
   */
  title?: React.ReactNode;
  /**
   * 应用栏可操作元素
   */
  actionItems?: React.ReactNode;
  /**
   * 滚动目标元素
   */
  targetScroll?: HTMLElement | Window | React.RefObject<HTMLElement>;
  /**
   * 给根元素指定自定义样式
   */
  style?: React.CSSProperties;
  /**
   * 给根元素指定自定义类名
   */
  className?: string;
}
