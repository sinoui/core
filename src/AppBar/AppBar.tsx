import styled, { css } from 'styled-components';
import AppBarTitle from '@sinoui/core/AppBarTitle';
import AppBarActions from '@sinoui/core/AppBarActions';
import NavigationIcon from '@sinoui/core/NavigationIcon';

/**
 * AppBar 顶部应用栏容器
 */
export interface Props {
  /**
   * 突出模式
   */
  prominent?: boolean;
  /**
   * 密集模式
   */
  dense?: boolean;
  /**
   * 固定模式
   */
  fixed?: boolean;
  /**
   * 收缩模式
   */
  short?: boolean;
}

/**
 * 常规 appBar
 */
const regularCss = css`
  height: 64px;
  padding: 8px 12px;
  align-items: center;
  box-sizing: border-box;

  @media (max-width: 599px) {
    height: 56px;
    padding: 4px;
  }
`;

/**
 * 突出 appBar
 */
const prominentCss = css`
  height: 128px;
  align-items: flex-start;
  padding: 8px 12px;
  box-sizing: border-box;

  & ${AppBarTitle} {
    align-self: flex-end;
    padding-bottom: 2px;
  }

  @media (max-width: 599px) {
    padding: 4px;
    & ${AppBarTitle} {
      padding-bottom: 6px;
    }
  }
`;

/**
 * dense 紧凑appbar
 */
const denseCss = css<Props>`
  height: ${({ prominent }) => (prominent ? '96px' : '48px')};
  padding: 0 ${({ prominent }) => (prominent ? '12px' : '4px')};

  & ${AppBarTitle} {
    padding-bottom: ${({ prominent }) => (prominent ? '9px' : 0)};
  }
`;

/**
 * 收缩状态的appbar
 */

const shortCss = css`
  width: 112px;
  height: 56px;
  padding: 4px;
  border-radius: 0 0 24px 0;
  & ${AppBarTitle} {
    display: none;
  }

  & ${NavigationIcon} {
    margin-right: 0;
  }

  & ${AppBarActions} {
    margin-left: 0;
    & > *:not(:last-child) {
      display: none;
    }
  }
`;

/**
 * fixed 为true 固定样式
 */
const fixedCss = css`
  position: fixed;
`;

const StyledAppBar = styled.div<Props>`
  position: relative;
  display: flex;
  width: 100%;
  color: ${({ theme }) => theme.palette.primary.contrastText};
  box-shadow: ${({ theme }) => theme.shadows[4]};
  background-color: ${({ theme }) => theme.palette.primary.main};
  ${({ prominent }) => (prominent ? prominentCss : regularCss)}
  ${({ dense }) => dense && denseCss}
  ${({ fixed }) => fixed && fixedCss}
  ${({ short }) => short && shortCss}
  & > ${AppBarTitle} {
    flex: 1;
    margin-left: 20px;
  }
  
`;
export default StyledAppBar;
