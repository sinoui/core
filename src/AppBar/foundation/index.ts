import TopAppBarFoundation from './TopAppBarFoundation';
import FixedFoundation from './FixedFoundation';
import ShortFoundation from './ShortFoundation';
import FixedProminentFoundation from './FixedProminentFoundation';
import ShortCollapsedFoundation from './ShortCollapsedFoundation';

export {
  TopAppBarFoundation,
  FixedFoundation,
  ShortFoundation,
  FixedProminentFoundation,
  ShortCollapsedFoundation,
};
