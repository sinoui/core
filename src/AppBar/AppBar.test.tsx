/**
 * @jest-environment jsdom
 */
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import AppBar from '@sinoui/core/AppBar';
import { render } from '@testing-library/react';

/**
 * 顶部应用栏AppBar组件 单元测试
 */

describe('快照测试', () => {
  it('标准模式', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <AppBar />
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('突出模式', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <AppBar prominent />
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('密集模式', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <AppBar dense />
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('收缩模式', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <AppBar dense />
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('固定模式', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <AppBar fixed />
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

it('className', () => {
  const { queryByTestId } = render(
    <ThemeProvider theme={defaultTheme}>
      <AppBar className="x-custom-app-bar" data-testid="appbar" />
    </ThemeProvider>,
  );

  expect(queryByTestId('appbar')).toHaveClass('x-custom-app-bar');
  expect(queryByTestId('appbar')).toHaveClass('sinoui-app-bar');
  expect(queryByTestId('appbar')).toHaveClass('sinoui-fixed');
});
