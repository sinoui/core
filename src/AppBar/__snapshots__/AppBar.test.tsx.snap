// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照测试 固定模式 1`] = `
<div
  className="AppBar__StyledAppBar-qqgfwp-0 khnbOp"
/>
`;

exports[`快照测试 密集模式 1`] = `
<div
  className="AppBar__StyledAppBar-qqgfwp-0 bebudo"
/>
`;

exports[`快照测试 收缩模式 1`] = `
<div
  className="AppBar__StyledAppBar-qqgfwp-0 bebudo"
/>
`;

exports[`快照测试 标准模式 1`] = `
<div
  className="AppBar__StyledAppBar-qqgfwp-0 htkjPv"
/>
`;

exports[`快照测试 突出模式 1`] = `
<div
  className="AppBar__StyledAppBar-qqgfwp-0 kuReVa"
/>
`;
